Foundation/NSObjCRuntime.swift

```diff
@@ -266,22 +266,143 @@ public func NSStringFromClass(_ aClass: AnyClass) -> String {
     neither stable nor human-readable.
  */
 public func NSClassFromString(_ aClassName: String) -> AnyClass? {
-    let aClassNameWithPrefix : String
-    let components = aClassName._bridgeToObjectiveC().components(separatedBy: ".")
-    
-    switch components.count {
-    case 1:
-        guard !aClassName.hasPrefix("_Tt") else {
-            NSLog("*** NSClassFromString(\(aClassName)): cannot yet decode mangled class names")
+    switch aClassName {
```

Comment:
I have an alternate solution, based on a workaround in this comment: 
https://bugs.swift.org/browse/SR-381?focusedCommentId=11290&page=com.atlassian.jira.plugin.system.issuetabpanels:comment-tabpanel#comment-11290


The `NSClassFromString()` seems to fail for any Class that doesn't directly conform to a protocol, so I added a dummy protocol and made the Classes that fail the tests conform to it.
See https://github.com/spevans/swift-corelibs-foundation/commit/99fa8153c830b5799eee10cf4bbeecd77eb444d7

Note that there are probably some other classes we need to fix, I just fixed up the ones that made `TestFoundation` fail when built for release mode running under Instruments.


https://bugs.swift.org/browse/SR-3333 is tracking this issue.